### PR TITLE
Adding plan_codes to coupon creation

### PR DESCRIPTION
**Pull request by eodonohoe**

### CODE REVIEW

**Comment by bhelx** [comment]
@eodonohoe Thanks a lot for contributing. I'll have a look at this in the morning.

File: src/main/java/com/ning/billing/recurly/model/Coupon.java [hunk 1]
```diff
@@ -96,6 +99,20 @@
...
+        return planCodes;
+    }
+
+    public void setPlanCodes(List<String> planCodes) {
+        this.planCodes = planCodes;
+    }
+
+    @XmlElement( name="plan_code" )
+    @XmlElementWrapper( name="plan_codes" )
+    private List<String> planCodes;
```
Comment (pierre) [inline src/main/java/com/ning/billing/recurly/model/Coupon.java]: Could you enhance existing tests for this?

Comment (eodonohoe) [inline src/main/java/com/ning/billing/recurly/model/Coupon.java]: Sure thing.  Added.

**Comment by bhelx** [comment]
It seems the tests in travis aren't picking up the subdomain from the environment.

```
java.util.concurrent.ExecutionException: java.net.ConnectException: https://.recurly.com:443
```

**Comment by eodonohoe** [comment]
Its supposed to pick that up from the repository settings correct?

**Comment by eodonohoe** [comment]
This may be the reason.
From: https://docs.travis-ci.com/user/encryption-keys/

_> Please note that encrypted environment variables are not available for pull requests from forks._

**Comment by bhelx** [comment]
Hmm, this has not been a problem before. @pierre have you noticed this?

**Comment by eodonohoe** [comment]
I do see that between build #69 and #71 (during merge of Pull Request #108) the way the keys were passed changed.  I don't see any outside forks merged in since then.

**Comment by bhelx** [comment]
I'd be happy to just confirm the tests locally if @pierre is cool with that

File: src/test/java/com/ning/billing/recurly/TestRecurlyClient.java [hunk 2]
```diff
@@ -17,10 +17,7 @@
 
 package com.ning.billing.recurly;
 
-import java.util.ArrayList;
-import java.util.HashSet;
-import java.util.List;
-import java.util.Set;
+import java.util.*;
```
Comment (pierre) [inline src/test/java/com/ning/billing/recurly/TestRecurlyClient.java]: We tend to avoid wildcard imports.

**Comment by pierre** [comment]
> Hmm, this has not been a problem before. @pierre have you noticed this?

Yup - PR on forks won't work for security reasons. I usually just get the branch and run the tests manually.

File: src/test/java/com/ning/billing/recurly/TestRecurlyClient.java (hunk 2 above)
Comment (eodonohoe) [inline src/test/java/com/ning/billing/recurly/TestRecurlyClient.java]: Fixed

**Comment by pierre** [comment]
👍

**Comment by bhelx** [comment]
Thanks @eodonohoe

I'm just going to run these tests locally

**Comment by bhelx** [comment]
:+1:

**Pull request merged**